code/ui/generalhud/menu/Menu.Settings.cs

```diff
@@ -0,0 +1,139 @@
+
+                if (Local.Client.HasPermission("serversettings"))
+                {
+                    SettingsTabs.AddTab("Server", InitServerSettings, Utils.Realm.Server);
+                }
+
+                CreateSettingsButtons(menuContent);
+            }, "Settings", "settings");
+        }
+
+        private void CreateClientSettings(PanelContent tabContent)
+        {
+            Panel languagePanel = tabContent.Add.Panel("language");
+
+            languagePanel.Add.Label("Language:");
+
+            Dropdown languageSelection = languagePanel.Add.Dropdown();
+
+            foreach (Language language in TTTLanguage.Languages.Values)
+            {
+                languageSelection.AddOption(language.Data.Name, language.Data.Code);
+            }
+
+            languageSelection.OnSelectOption = (option) =>
+            {
+                ConsoleSystem.Run("ttt_language", (string) option.Data);
+            };
+
+            languageSelection.SelectByData(SettingsManager.Instance.Language);
+        }
+
+        private void InitServerSettings(PanelContent tabContent)
+        {
+            ServerSettingsTabContent = tabContent;
+
+            tabContent.Add.Label("Loading...");
+
+            ConsoleSystem.Run("ttt_serversettings_request");
+        }
+
+        internal void CreateServerSettings(PanelContent tabContent, ServerSettings serverSettings)
+        {
+            tabContent.DeleteChildren(true);
+
+            tabContent.Add.Label($"Sprint enabled?");
+            Switch sw = tabContent.Add.Switch("sprint", serverSettings.IsSprintEnabled);
+
+            sw.AddEventListener("onchange", (panelEvent) =>
+            {
+                serverSettings.IsSprintEnabled = !serverSettings.IsSprintEnabled;
+
+                // TODO Can be improved to avoid syncing EVERYTHING
+                // send settings to server
+                ConsoleSystem.Run("ttt_serversettings_send", SettingFunctions.GetJSON<ServerSettings>(serverSettings, true));
+            });
+        }
+    }
+}
+
+namespace TTTReborn.Player
+{
+    using TTTReborn.UI.Menu;
+
+    public partial class TTTPlayer
+    {
+        [ServerCmd(Name = "ttt_serversettings_request")]
+        public static void RequestServerSettings()
+        {
+            if (!ConsoleSystem.Caller.HasPermission("serversettings"))
+            {
+                return;
+            }
+
+            ClientSendServerSettings(To.Single(ConsoleSystem.Caller), SettingFunctions.GetJSON<ServerSettings>(SettingsManager.Instance as ServerSettings, true));
+        }
+
+        [ServerCmd(Name = "ttt_serversettings_send")]
+        public static void SendServerSettings(string serverSettingsJson)
+        {
+            if (!ConsoleSystem.Caller.HasPermission("serversettings"))
+            {
+                return;
+            }
+
+            ServerSettings serverSettings = SettingFunctions.GetSettings<ServerSettings>(serverSettingsJson);
```

Comment:
This line causes the client's `SettingsManager.Instance` to be overwritten by the  server data (e.g. server language). I can't find the reason, maybe it's an issue with local and remote branch merging of sbox (the master branch that is used as game base still has the Instance field that is overwritten when a new Setting is created).